src/activityLog.ts

```diff
@@ -0,0 +1,21 @@
+import * as path from 'path';
+import * as os from 'os';
+import * as yaml from 'js-yaml';
+import OutputFile from './outputFile';
+
+export default class ActivityLog {
+    private outputFile = new OutputFile('ActivityLog.yaml', path.join(os.tmpdir(), 'VSLiveShareActivityLogs'));

```

Comment:
I wonder if instead of a yaml file, we should just write this as a ["JSON Lines"](http://jsonlines.org/) log? Since JSON is a much more common log format, that would probably make it easier to consume if other's wanted to process the log. And the other benefit of JSON Lines is that you could process the log as a stream, as opposed to needing to parse the entire file. Maybe that isn't a big deal, but in theory, one of these logs could get pretty big, depending on the length of the session and the number of edits?

Reply:
Looks like there are fewer JSON Lines parsers than yaml parsers. Also, if we ever want to add metadata to the log file, it'd be slightly more unnatural with JSON Lines.

However, it's an easy change if you feel like it's better.